### PR TITLE
Allow fractional HiDPI to start, but warn about non support.

### PR DESCRIPTION
**Pull request by Moar-Chan**
Allow fractional to start, but warn about non support.

### CODE REVIEW

**Comment by Moar-Chan** [comment]
Sorry if i did somthing wrong (This might be my first ever pull request)

**Comment by lcnr** [comment]
Thank you for the PR :heart: 

Can you replace `println` with `warn` though? https://docs.rs/log/0.4.8/log/macro.warn.html
`println` can't be silenced and does not mention its origin.

**Comment by Moar-Chan** [comment]
Why is it failing?

**Comment by lcnr** [comment]
I use `cargo fmt` in this repository and check this automatically.

You can run the command locally to fix this

**Comment by Moar-Chan** [comment]
ah ok

File: src/backend/mod.rs [hunk 1]
```diff
@@ -133,7 +133,10 @@ impl Backend {
         if dpi < 0.5 {
             bug!("unexpected dpi: {}", dpi);
         } else if dpi.fract().min(1.0 - dpi.fract()) > std::f64::EPSILON {
-            bug!("fractional HiDPI scaling is not yet supported: {}", dpi);
+            warn!(
+                "fractional HiDPI scaling is not yet supported. but may work! (dpi: {})",
```
Comment (lcnr) [inline src/backend/mod.rs:137]:
```suggestion
                "fractional HiDPI scaling is not yet fully supported! (dpi: {})",
```

Comment (Moar-Chan) [reply src/backend/mod.rs:137]: True

Comment (lcnr) [reply src/backend/mod.rs:137]: Once you've changed this I am going to merge your PR :+1: I don't want to accept your PR and then adapt it later as that would be a waste of both of our time.

Thanks for your interest and readiness to help :heart:

**Pull request merged**